File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, time int) int {
    if time == 0 {
        time = 2
    }
    return time * len(layers)
}
// TODO: define the 'Quantities()' function
func Quantities(layers []string) (int, float64){
    noodles, sauce := 0, 0
    for i, _ := range layers {
        if layers[i] == "noodles" {
            noodles += 1
        } else if layers[i] == "sauce" {
            sauce += 1
        }
    }
    return noodles * 50, float64(sauce) * 0.2
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friend_recipe []string, your_recipe []string) {
    your_recipe[len(your_recipe) - 1] = friend_recipe[len(friend_recipe) - 1]
}
// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(amount []float64, portions int) []float64 {
    new_amount := append([]float64{}, amount...)
    for i, _ := range new_amount {
        new_amount[i] *= float64(portions) / 2
    }
    return new_amount
}
// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
// 
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more 
// functionality.

File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/*
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */

export function cookingStatus(time) {
  if (time === 0) {
    return "Lasagna is done.";
  }
  if (time > 0) {
    return "Not done, please wait.";
  }
  return "You forgot to set the timer.";
}

export function preparationTime (layers, time) {
  if (!time) {
    time = 2;
  }
  return time * layers.length;
}

export function quantities(layers) {
  let noodles = 0;
  let sauce = 0;
  for (let i in layers) {
    if (layers[i] == "noodles") {
      noodles += 1;
    } else if (layers[i] == "sauce") {
      sauce += 1;
    }
  }

  return {
    noodles: noodles * 50, 
    sauce: sauce * 0.2
  };
}

export function addSecretIngredient(friend_recipe, your_recipe) {
  const secret = friend_recipe[friend_recipe.length - 1];
  
  your_recipe.push(secret);
}

export function scaleRecipe(recipe, portions) {
  const scaled = {};
  const factor = portions / 2;

  for (const ingredient in recipe) {
    scaled[ingredient] = recipe[ingredient] * factor;
  }

  return scaled;
}